src/page-view.ts

```diff
@@ -145,22 +155,45 @@ export class PageViewManager {
         window?.removeEventListener('resize', this._updateScrollData)
     }
 
+    _scrollElement(): Element | null | undefined {
+        if (this._instance.config.scroll_root_selector) {
+            const selectors = _isArray(this._instance.config.scroll_root_selector)
+                ? this._instance.config.scroll_root_selector
+                : [this._instance.config.scroll_root_selector]
+            for (const selector of selectors) {
+                const element = window?.document.querySelector(selector)
+                if (element) {
+                    return element
+                }
+            }
+            return undefined
+        } else {
+            return window?.document.documentElement
+        }
+    }
+
     _scrollHeight(): number {
-        return window
-            ? Math.max(0, window.document.documentElement.scrollHeight - window.document.documentElement.clientHeight)
-            : 0
+        const element = this._scrollElement()
```

Comment:
similar nit on these helpers, we could lookup the element once and pass it into each of these helpers... 

(may be so fast it doesn't matter :))